File: ReSharper.FSharp/src/FSharp.Psi/src/Metadata/ZoneMarker.cs
using JetBrains.Application.BuildScript.Application.Zones;
using JetBrains.Platform.RdFramework;

namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Metadata
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<IRdFrameworkZone>, IRequire<ISinceClr4HostZone>
  {
  }
}

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/LanguageService/ZoneMarker.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.LanguageService

open JetBrains.Application.BuildScript.Application.Zones
open JetBrains.Platform.RdFramework
open JetBrains.ReSharper.Feature.Services
open JetBrains.ReSharper.Feature.Services.Breadcrumbs
open JetBrains.ReSharper.Feature.Services.Daemon

[<ZoneMarker>]
type ZoneMarker() =
    interface IRequire<DaemonZone>
    interface IRequire<IBreadcrumbsZone>
    interface IRequire<ICodeEditingZone>
    interface IRequire<IRdFrameworkZone>
    interface IRequire<ISinceClr4HostZone>
